stripe-ui-core/src/main/java/com/stripe/android/uicore/elements/DateConfig.kt

```diff
@@ -63,17 +63,23 @@ class DateConfig : TextFieldConfig {
             month1Based: Int,
             twoDigitYear: Int,
             current1BasedMonth: Int,
-            currentYear: Int
+            currentYear: Int,
         ): TextFieldState {
             val twoDigitCurrentYear = currentYear % 100
 
-            return if ((twoDigitYear - twoDigitCurrentYear) < 0) {
-                Error.Invalid(R.string.stripe_invalid_expiry_year)
-            } else if ((twoDigitYear - twoDigitCurrentYear) > 50) {
-                Error.Invalid(R.string.stripe_invalid_expiry_year)
-            } else if ((twoDigitYear - twoDigitCurrentYear) == 0 && current1BasedMonth > month1Based) {
-                Error.Invalid(R.string.stripe_invalid_expiry_month)
-            } else if (month1Based !in 1..12) {
+            val isExpiredYear = (twoDigitYear - twoDigitCurrentYear) < 0
+            val isYearTooLarge = (twoDigitYear - twoDigitCurrentYear) > 50
+
+            val isExpiredMonth = (twoDigitYear - twoDigitCurrentYear) == 0 && current1BasedMonth > month1Based
+            val isMonthInvalid = month1Based !in 1..12
+
+            return if (isExpiredYear) {
+                Error.Invalid(R.string.stripe_invalid_expiry_year, preventMoreInput = true)
+            } else if (isYearTooLarge) {
+                Error.Invalid(R.string.stripe_invalid_expiry_year, preventMoreInput = true)
+            } else if (isExpiredMonth) {
+                Error.Invalid(R.string.stripe_invalid_expiry_month, preventMoreInput = true)
```

Comment:
What happens if the date is 2023/Oct/1? Then the user will not be able to add a zero to the month?

Reply:
What do you mean? If the user enters `11`, then the existing logic will transform it to `11 / `. If they type `011`, the existing logic transforms it to `01 / 1` and they can enter the final digit of the year.

Reply:
I am mainly focused on the `isExpiredMonth`. Just checking any edge cases that would make it not possible to enter in a full month. Lgtm.